### PR TITLE
XW-1508 | Reload page for unsupported links

### PR DESCRIPTION
**Pull request by rogatty**
## Links

* https://wikia-inc.atlassian.net/browse/XW-1508

## Description

Disallow in-app links to routes other than main pages, wiki pages, article edit and article photo upload.
You can use https://regex101.com/r/iF5sP7/1 if you want to try to break the regex.

## Reviewers

@Wikia/x-wing

### CODE REVIEW

**Comment by vforge** [comment]
LGTM

File: front/main/app/initializers/link-to.js [hunk 1]
```diff
@@ -36,6 +37,13 @@ export function initialize() {
 				});
 			}
 
+			// XW-1492 if we don't support the link then reload the page and go through VCL
+			if (event.target.href && !isSpaLink(event.target.href, window.location.origin)) {
+				window.location.reload(event.target.href);
```
Comment (bognix) [inline front/main/app/initializers/link-to.js]: `window.location.reload` takes as an argument `true` or `false` - depending if you want to perform regular reload or `forceReload`
https://developer.mozilla.org/en-US/docs/Web/API/Location/reload

It doesn't take as an argument `href`.

What I think you should use here in `window.location.assign`. It creates new entry in History. We already use it in couple of places in Mercury.

File: front/main/app/utils/spa-link.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+/**
+ * @param {String} href
+ * @param {String} origin
+ * @returns {Boolean}
+ */
+export function isSpaLink(href, origin) {
+	// XW-1492
+	// We won't support other routes than these ones in the app custom version
+	const supportedRoutes = new RegExp('^' + origin + '/(wiki/.*|main/(section|category)/.+|$)');
```
Comment (bognix) [inline front/main/app/utils/spa-link.js]: we didn't remove `Edit` functionality from `skinny-mercury`? Based on this regex edit page would be recognized as SPA link

**Comment by bognix** [comment]
Please resolve the comment regarding `window.location.reload`.
Besides that - LGTM

File: front/main/app/utils/spa-link.js (hunk 2 above)
Comment (rogatty) [inline front/main/app/utils/spa-link.js]: Article edit stays with us, see https://github.com/Wikia/mercury-article-experiment/blob/XW-1508/front/main/tests/unit/utils/spa-link-test.js#L8 for the list of currently supported routes.

**Pull request merged**